developer-workflow/extension-modules.rst

```diff
@@ -5,13 +5,579 @@
+       ...
+     </ItemGroup>
+
+* Update :cpy-file:`PCbuild/pythoncore.vcxproj.filters`:
+
+  .. code-block:: xml
+     :caption: :cpy-file:`PCbuild/pythoncore.vcxproj.filters`
+     :emphasize-lines: 4-6, 13-18
+
+     <!-- group with header files ..\Modules\<MODULE>.h -->
+     <ItemGroup>
+       ...
+       <ClInclude Include="..\Modules\_foo\_foomodule.h">
+         <Filter>Modules\_foo</Filter>
+       </ClInclude>
+       ...
+     </ItemGroup>
+
+     <!-- group with source files ..\Modules\<MODULE>.c -->
+     <ItemGroup>
+       ...
+       <ClCompile Include="..\Modules\_foo\_foomodule.c">
+         <Filter>Modules\_foo</Filter>
+       </ClCompile>
+       <ClCompile Include="..\Modules\_foo\helper.c">
+         <Filter>Modules\_foo</Filter>
+       </ClCompile>
+       ...
+     <ItemGroup>
+
+.. tip::
+
+   Header files use ``<ClInclude>`` tags, whereas
+   source files use ``<ClCompile>`` tags.
+
+Updating :cpy-file:`!Modules/Setup.{bootstrap,stdlib}.in`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Depending on whether the extension module is required to get a functioning
+interpreter or not, we update :cpy-file:`Modules/Setup.bootstrap.in` or
+:cpy-file:`Modules/Setup.stdlib.in`. In the former case, the extension
+module is necessarily built as a built-in extension module.
+
+.. tip::
+
+   For accelerator modules, :cpy-file:`Modules/Setup.stdlib.in` should be
+   preferred over :cpy-file:`Modules/Setup.bootstrap.in`.
+
+For built-in extension modules, update :cpy-file:`Modules/Setup.bootstrap.in`
+by adding the following line after the ``*static*`` marker:
+
+.. code-block:: text
+   :caption: :cpy-file:`Modules/Setup.bootstrap.in`
+   :emphasize-lines: 3
+
+   *static*
+   ...
+   fastfoo _foo/_foomodule.c _foo/helper.c
+   ...
+
+For other extension modules, update :cpy-file:`Modules/Setup.stdlib.in`
+by adding the following line after the ``*@MODULE_BUILDTYPE@*`` marker
+but before the ``*shared*`` marker:
+
+.. code-block:: text
+   :caption: :cpy-file:`Modules/Setup.stdlib.in`
+   :emphasize-lines: 3
+
+   *@MODULE_BUILDTYPE@*
+   ...
+   @MODULE_FASTFOO_TRUE@fastfoo _foo/_foomodule.c _foo/helper.c
+   ...
+   *shared*
+
+The ``@MODULE_<NAME>_TRUE@<name>`` marker expects ``<NAME>`` to be the
+upper-cased module name ``<name>``. If the extension module must be built
+as a *shared* module, put the ``@MODULE_FASTFOO_TRUE@fastfoo`` line after
+the ``*shared*`` marker:
+
+.. code-block:: text
+   :caption: :cpy-file:`Modules/Setup.stdlib.in`
+   :emphasize-lines: 4
+
+   ...
+   *shared*
+   ...
+   @MODULE_FASTFOO_TRUE@fastfoo _foo/_foomodule.c _foo/helper.c
+
+Compiling the CPython project
+-----------------------------
+
+Now that everything is in place, it remains to compile the project:
+
+.. code-block:: shell
+
+   make regen-configure
+   ./configure
+   make regen-all
+   make regen-stdlib-module-names
+   make
+
+.. tip::
+
+   Use ``make -j12`` to speed-up compilation if you have enough CPU cores.
```

Comment:
Quick question but should I say ``make -j12``, ``make -j`` or ``make -j4``? There is an opened PR about this https://github.com/python/devguide/pull/1087 but I put 12 because I don't like to wait (also because I have 32 in total) but should I go for something like `-j8` instead (or even a more conservative `-j4`?)

Reply:
Recommending `-j` and `-jN` instead.